e2e/testdata/root-sync-helm-chart-version-range-cr.yaml

```diff
@@ -0,0 +1,53 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: configsync.gke.io/v1beta1
+kind: RootSync
+metadata:
+  name: root-sync
+  namespace: config-management-system
+spec:
+  sourceFormat: unstructured
+  sourceType: helm
+  helm:
+    repo: https://charts.bitnami.com/bitnami
+    chart: wordpress
+    version: ^15.0.0
```

Comment:
This syntax is natively supported by helm CLI.

See [https://github.com/Masterminds/semver#hyphen-range-comparisons](https://www.google.com/url?q=https://github.com/Masterminds/semver%23hyphen-range-comparisons) for more details on the syntax

Reply:
Will a new version trigger an update automatically? Is there a way test that? 

FYI - Because the update is triggered without changing R*Sync spec, this is going to make `status.observedGeneration` less useful. Tagging @sdowell given that we discussed this earlier.

Reply:
> Will a new version trigger an update automatically? Is there a way test that?

I tried it manually and yes, it does an automatic update. 

I'll see if I can add a test. The test would presumably involve pushing a new chart version to the test registry and then deleting it after, which I think is possible, but I will have to look more into it.

Reply:
Actually, do you know if the test env has helm CLI on it? I think that would be needed for adding a test

Reply:
Assuming you are referring to the e2e test env, [yes it has helm CLI](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/build/test-e2e-go/gke/Dockerfile#L41). Looks like the e2e test runner GSA currently does not have permission to delete images, but that could be added. Why does it need to delete images? For test cleanup?

This would be another use case where using an in-cluster/ephemeral/hermetic registry would be preferred. That shouldn't be a blocker for you here, just wanted to flag it

Reply:
> Why does it need to delete images? For test cleanup?

Yes for test cleanup; this test would need to push a new tag and clean up the tag afterwards. Since we can add the permissions, seems like there are no blockers so I will try to add the test.

I agree hermetic registry would be a better solution in the long term.

Reply:
Pushed the new test.

/hold (until the e2e test env GSA to have the right permissions)

Reply:
@natasha41575 Thanks for adding the test. Would it trigger an upgrade (to the latest version) when a version range is specified, too?

Reply:
Yes, multiple instances of this test may be running concurrently (e.g. for the different GKE channels) - and that's something we need to account for

Reply:
@janetkuo What would you like to do here? Should we remove the part of the test I added for automatic updates and merge the PR with just the other tests for now, or should we block on rewriting these tests to be more hermetic/not use an external AR?

Reply:
I'd prefer we make the tests more hermetic if possible. It will likely block us from adding more tests in the future.

Reply:
This will be unblocked by https://github.com/GoogleContainerTools/kpt-config-sync/pull/636. When that is merged I will rebase this PR